### ISSUE TITLE
Leaked connection to  https://sessions.bugsnag.com/

### ISSUE THREAD

**Issue by atexit** [bugsnag/bugsnag-android#232]
### Expected behavior
Session tracking connection properly closed. 

### Observed behavior
OkHttp leak detection found the following while I was running our integration tests.

`W/OkHttpClient: A connection to https://sessions.bugsnag.com/ was leaked. Did you forget to close a response body?`

Not sure this is a real issue, but figured I had better report it to you 

### Steps to reproduce
* We use OkUrlFactory to use OkHttp even for legacy parts of our client, so we set the URL stream handler factory. I know it's deprecated, but see it as a way to get connection leak tracking in this case: 
       `URL.setURLStreamHandlerFactory(new OkUrlFactory(okHttpClient));`
* Use session autotracking.

### Version
4.2.2

**Comment by fractalwrench** [comment]
Thanks for the report @atexit, is this something you're seeing consistently? I've set the StreamHandlerFactory using v3.6.0 of OkHttp as follows, and couldn't immediately see any error message:

```
URL.setURLStreamHandlerFactory(new OkUrlFactory(new OkHttpClient.Builder().build()));
```

I'll double-check how our HttpClient closes itself.

**Comment by atexit** [comment]
I can't say for sure in production. So this may be a side effect of espresso/instrumentation test or our OkHttp setup.

**Comment by atexit** [comment]
We've been seeing a lot of timeouts in networking this weekend, after releasing with the 4.2.2  Bugsnag. There might be something here after all. Will try to narrow it down.

**Comment by MoNTE48** [comment]
I also see this in the logs with the latest version.

**Comment by johnkiely1** [comment]
Hi @MoNTE48 
Do you have any more information you could share on this? Are you able to reproduce this behaviour?

**Comment by smn-snkl** [comment]
Seeing the same error in our current setup. It happens on startup right after calling bugsnag.start() via the RN SDK.

**Comment by phillipsam** [comment]
A fix for this has now been released in [v5.5.1](https://github.com/bugsnag/bugsnag-android/releases/tag/v5.5.1).
